yt/frontends/gadget/io.py

```diff
@@ -67,6 +67,19 @@ def _read_particle_coords(self, chunks, ptf):
                 yield ptype, (x, y, z), hsml
             f.close()
 
+    def _datafile_has_ptf(self, data_file, ptf):
+        """
+        checks a data_file object to see if any particle types in ptf dict
+        have a non-zero particle count in this file. use to avoid opening
+        files if we know the particle counts a priori (the index construction
+        does not differentiate by particle type).
+        """
+        for ptype, _ in sorted(ptf.items()):
```

Comment:
I know you didn't originate this, but we can change this to just be `for ptype in sorted(ptf)`